cw/cw305/simple_capture_traces.py

```diff
@@ -6,137 +6,145 @@
 import binascii
 from Crypto.Cipher import AES
 import numpy as np
-import sys
 import time
 from tqdm import tqdm
 import yaml
+import re
 
 import chipwhisperer as cw
 
 from util import device
 from util import plot
-from util import spiflash
-
-
-def initialize_capture(device_cfg):
-  """Initialize capture."""
-  fw_programmer = spiflash.SAM3UProgrammer(device_cfg['fw_bin'])
-
-  ot = device.OpenTitan(
-    fw_programmer,
-    device_cfg['fpga_bitstream'],
-    device_cfg['pll_frequency'],
-    device_cfg['baudrate'])
-  print(f'Scope setup with sampling rate {ot.scope.clock.adc_rate} S/s')
-  # Ping target
-  print('Reading from FPGA using simpleserial protocol.')
-  version = None
-  ping_cnt = 0
-  while not version:
-    if ping_cnt == 3:
-      raise RuntimeError(f'No response from the target (attempts: {ping_cnt}).')
-    ot.target.write('v' + '\n')
-    ping_cnt += 1
-    time.sleep(0.5)
-    version = ot.target.read().strip()
-  print(f'Target simpleserial version: {version} (attempts: {ping_cnt}).')
-  return ot
+from vendor.lowrisc_opentitan_util import cw_spiflash
+
+
+def initialize_capture(device_cfg, capture_cfg):
+    """Initialize capture."""
+    # Extract target board type from bitstream name.
+    if re.search('cw310', device_cfg['fpga_bitstream']):
+        board = 'CW310'
+    else:
+        board = 'CW305'
```

Comment:
I think `board` should ideally be an enum, can you add a TODO for that?

Also:
```suggestion
    m = re.search('cw305|cw310', device_cfg['fpga_bitstream'])
    if m:
        if m.group() == 'cw305':
            board = 'CW305'
        else:
            assert m.group() == 'cw310'
            board = 'CW310'
    else:
        raise ValueError('Could not infer target board type from bitstream name')
```

Reply:
Created an issue, and switched to your suggested code. Thanks for your review @alphan !